### PR TITLE
gh-112536: Add more TSAN tests

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-112536

### CODE REVIEW

**Comment by corona10** [comment]
Test list from: https://github.com/swtaarrs/cpython/commit/1fe9165e28232d34512a8f476d598d62ba5e4b55#diff-590c02490d066378045dd48d62e86dc85d3fb4ad934b6a2b2b0b1112e59eaefb

File: Lib/test/libregrtest/tsan.py [hunk 1]
```diff
@@ -3,19 +3,24 @@
 
 TSAN_TESTS = [
     'test_code',
+    'test_compileall',
     'test_enum',
     'test_functools',
+    'test_fork1',
```
Comment (pitrou) [inline Lib/test/libregrtest/tsan.py]: Why add this test? fork() doesn't work with TSAN.

File: Lib/test/libregrtest/tsan.py [hunk 2]
```diff
@@ -3,19 +3,24 @@
 
 TSAN_TESTS = [
     'test_code',
+    'test_compileall',
```
Comment (pitrou) [inline Lib/test/libregrtest/tsan.py]: Why this one? AFAIR it's long and shouldn't really stress any low-level threading issue.

File: Lib/test/libregrtest/tsan.py [hunk 3]
```diff
@@ -3,19 +3,24 @@
...
     'test_enum',
     'test_functools',
+    'test_fork1',
     'test_httpservers',
     'test_imaplib',
     'test_importlib',
     'test_io',
     'test_logging',
+    'test_pickle',
```
Comment (pitrou) [inline Lib/test/libregrtest/tsan.py]: Similarly, why this one?

**Comment by corona10** [comment]
> test_pickle
> test_compileall
> test_fork1

@pitrou 
Those tests are currently not passed with PYTHON_GIL =0, but if they are not working with TSAN test, no worth adding it.
Thank you for pointing out :)

File: Lib/test/libregrtest/tsan.py [hunk 4]
```diff
@@ -10,12 +10,14 @@
     'test_importlib',
     'test_io',
     'test_logging',
+    'test_queue',
```
Comment (corona10) [inline Lib/test/libregrtest/tsan.py:13]: https://github.com/python/cpython/blob/86bc40dd414bceb3f93382cc9f670936de9d68be/Lib/test/test_queue.py#L165

File: Lib/test/libregrtest/tsan.py [hunk 5]
```diff
@@ -10,12 +10,14 @@
...
     'test_ssl',
     'test_syslog',
     'test_thread',
     'test_threadedtempfile',
     'test_threading_local',
     'test_threadsignals',
+    'test_weakref',
```
Comment (corona10) [inline Lib/test/libregrtest/tsan.py:21]: https://github.com/python/cpython/blob/86bc40dd414bceb3f93382cc9f670936de9d68be/Lib/test/test_weakref.py#L1837-L1866

**Review by pitrou — APPROVED** [review]

**Pull request merged**